Riot/ViewController/HomeViewController.m

```diff
@@ -591,7 +591,41 @@ - (IBAction)onPriorityButtonPressed:(id)sender
 
 - (IBAction)onLeaveButtonPressed:(id)sender
 {
-    [self leaveEditedRoom];
+    __weak typeof(self) weakSelf = self;
```

Comment:
Could you move this code inside `[RecentsViewController leaveEditedRoom]` so that all tabbed screens can take benefit on it?
If you swipe left on a room of a rooms list, the same set of buttons with the leave button is displayed.

Reply:
I tried to move the code inside **[self leaveEditedRoom]** but problem is **editedRoomId** get null and the reason is:
```
- (void)tableView:(UITableView*)tableView didEndEditingRowAtIndexPath:(NSIndexPath *)indexPath
{
    [self cancelEditionMode:isRefreshPending];
}
```
what do you suggest?

Reply:
@manuroe done!